### ISSUE TITLE
The search function frequently does not find notes that match the word.

### ISSUE THREAD

**Issue by WarriorMama777** [laurent22/joplin#8353]
## Issue
The search function frequently does not find notes that match the word.

## Environment

Joplin 2.10.19 (prod, win32)
portable

client ID: 
sync version: 3
profile version: 42
key chain support: No

Revision: 1100ab0

Platform: Windows10
Lang: Japanese
Encryption: No

## Steps to reproduce

1. set lang "Japanese"  (probably not needed for reproduction)
2. Import ENEX: Evernote export file (Import as Markdown)  (I don't know if it is necessary for reproduction.)
3. search by word
4. The search function frequently does not find notes that match the word.

▼The note I want to find. Word: "Tensor"
![2023_06_22__20h_59m_37s_flameshot](https://github.com/laurent22/joplin/assets/100090558/1ee95215-cc53-404a-a468-10560d00f0af)

▼search finds no notes
![y5425y24u5h5](https://github.com/laurent22/joplin/assets/100090558/e46a1756-decc-4ce6-ba3f-c575e46cdc69)

▼The note I'm looking for contains "Tensor" in both the title and the text.
![image](https://github.com/laurent22/joplin/assets/119816000/65525a3b-2159-4c05-bdfe-a67028e44ef9)

▼I found it in the command palette.
However, I've encountered word-note combinations that neither the search function nor the command palette finds.
![5y24525y2](https://github.com/laurent22/joplin/assets/100090558/d962036c-dbc0-4770-ab6b-d28c30d9190c)

## Reference

Related reports

- Issues · laurent22/joplin | https://github.com/laurent22/joplin/issues?page=1&q=is%3Aissue+is%3Aopen+search
- Search results for 'search' - Joplin Forum | https://discourse.joplinapp.org/search?q=search%20order%3Alatest
- Search optimization suggestions - Features - Joplin Forum | https://discourse.joplinapp.org/t/search-optimization-suggestions/7427/7
- [Joplin 1.7.11] search engine doesn't work accurate · Issue #4482 · laurent22/joplin | https://github.com/laurent22/joplin/issues/4482
- Confusing search - Support - Joplin Forum | https://discourse.joplinapp.org/t/confusing-search/7628
- Can not search chinese in titile - Support - Joplin Forum | https://discourse.joplinapp.org/t/can-not-search-chinese-in-titile/1361
- Joplin search function is ugly - Features - Joplin Forum | https://discourse.joplinapp.org/t/joplin-search-function-is-ugly/25824
- Shortcut for title search or tag search? - Lounge - Joplin Forum | https://discourse.joplinapp.org/t/shortcut-for-title-search-or-tag-search/13771/2
- It is recommended to add the notebook name to the search scope - Features - Joplin Forum | https://discourse.joplinapp.org/t/it-is-recommended-to-add-the-notebook-name-to-the-search-scope/24344
- Search note problem (can't find a note even with words from its title) - Support - Joplin Forum | https://discourse.joplinapp.org/t/search-note-problem-cant-find-a-note-even-with-words-from-its-title/577/17
- Search in notebook names - Features - Joplin Forum | https://discourse.joplinapp.org/t/search-in-notebook-names/7456/2
- Desktop: Go To Anything by body by anjulalk · Pull Request #2686 · laurent22/joplin | https://github.com/laurent22/joplin/pull/2686

**Comment by dpoulton-github** [comment]
According to the help page at Joplinapp.org, if you are searching for a word (such as `Tensor` in your second image, above) where it may form part of a larger word (such as `TensorRT`) then the search term is `tensor*`.

https://joplinapp.org/help/#searching

**Comment by WarriorMama777** [comment]
fulltext search by / worked for me.
(I Posted for anyone wandering the forums): https://discourse.joplinapp.org/t/post-for-those-having-trouble-with-joplin-search-not-working/31381

"Basic" being an option setting might feel uncomfortable for many people, but since it worked, I'll close it with this.

**Comment by WarriorMama777** [comment]
I reopened the issue for the following two reasons:

1. Plugins like "Quick Links" (roman-r-m/joplin-plugin-quick-links - https://github.com/roman-r-m/joplin-plugin-quick-links) do not support full-text search, resulting in an inability to detect notes.

2. It still feels uncomfortable that the default search method has accuracy issues. I believe it would be beneficial from a UX perspective to consider some improvements that involve making Basic Search (Full-Text Index Queries), which offers good accuracy without being fatally slow, the default option.

At the very least, having the ability to switch between different search methods through settings is a positive aspect. I have attached a mockup of a UI, where selection can be made through a dropdown menu under "Setting > General > Search Engine" (although I created it without fully understanding the technical backend considerations).

![Frame 1joplin_UI_03_](https://github.com/laurent22/joplin/assets/119816000/975c3159-8666-47b1-8be9-b78889f1aaae)
